src/watcher.ts

```diff
@@ -1,39 +1,61 @@
 import chokidar from 'chokidar';
 import { exec, ChildProcess } from 'child_process';
+import type Serverless from 'serverless';
 
-export type Config = {
+type ConfigItem = {
   path: string | string[];
-  command: string;
-}[];
+  command?: string;
+  hooks?: string | string[];
+};
+
+export type Config = ConfigItem[];
 
 export interface Watcher {
   start: () => void;
   stop: () => Promise<void>;
 }
 
-export const makeWatcher = (config: Config): Watcher => {
+export const makeWatcher = (config: Config, serverless: Serverless): Watcher => {
   const internalWatchers: chokidar.FSWatcher[] = [];
   const processes: { [pid: number]: ChildProcess } = {};
 
+  const runCommand = (command: string, eventType: string, eventPath: string) => new Promise<void>((resolve) => {
+    const env = {
```

Comment:
Note: (to self) Technically, tweaking the environment in which the command would get run in is a breaking change. However, the variables seem unique enough (supported by that they don't appear on GitHub before) that I reckon it's okay to release as a minor version.

Reply:
True. We could make this fully backwards-compatible by doing the spread other way around,  
```javascript
   WATCHER_EVENT_TYPE: eventType,
   WATCHER_EVENT_PATH: eventPath,
   ...process.env,
```
so that existing ones in env would override thse.  

But that could lead to another suprising behaviour if your `command` were to launch a _new_ serverless process that in turn runs these commands, then these new commands wouldn't get the variables here because they are already inherited from the parent.

Both scenarios, someone's code breaking because of the env variables and someone launching new serverless processes from the commands here, sound like very unlikely edge cases. But I feel that _if_ the latter were to happen it could be a pretty nasty footgun to debug.

Reply:
Agreed. I think the way it's implemented makes the most sense, and would be the way I'd expect it to work as a user.